standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/leader/HouseKeepingTasks.java

```diff
@@ -61,6 +62,11 @@ public List<MetastoreTaskThread> getRemoteOnlyTasks() throws Exception {
     Collection<String> taskNames =
         MetastoreConf.getStringCollection(configuration, MetastoreConf.ConfVars.TASK_THREADS_REMOTE_ONLY);
     for (String taskName : taskNames) {
+      if (AcidCompactionHouseKeeperService.class.getName().equals(taskName) &&
```

Comment:
move this logic to the service ifself

Reply:
Moved logic to service, my intention to keep it here is to avoid creating service itself when initiator is not enabled